engine/src/flutter/shell/common/shell.cc

```diff
@@ -1107,8 +1105,29 @@ void Shell::OnPlatformViewSetViewportMetrics(int64_t view_id,
 
   {
     std::scoped_lock<std::mutex> lock(resize_mutex_);
-    expected_frame_sizes_[view_id] =
-        DlISize(metrics.physical_width, metrics.physical_height);
+
+    bool has_given_constraints = (metrics.min_width_constraint != 0.0 ||
+                                  metrics.max_width_constraint != 0.0 ||
+                                  metrics.min_height_constraint != 0.0 ||
+                                  metrics.max_height_constraint != 0.0);
+
+    if (has_given_constraints) {
+      expected_frame_constraints_[view_id] = {
+          .min_width = metrics.min_width_constraint,
+          .max_width = metrics.max_width_constraint,
+          .min_height = metrics.min_height_constraint,
+          .max_height = metrics.max_height_constraint,
+
+      };
+    } else {
+      expected_frame_constraints_[view_id] = {
+          .min_width = metrics.physical_width,
+          .max_width = metrics.physical_width,
+          .min_height = metrics.physical_height,
+          .max_height = metrics.physical_height,
+      };
+    }
```

Comment:
If the embedder does not support content-sized views, it should pass tight constraints. If so, can we simplify this logic to just use the constraints? (Feel free to push back on this feedback if I'm missing something!)

```suggestion
    expected_frame_constraints_[view_id] = {
        .min_width = metrics.min_width_constraint,
        .max_width = metrics.max_width_constraint,
        .min_height = metrics.min_height_constraint,
        .max_height = metrics.max_height_constraint,
    };
```